pkg/net/libp2p/channel_manager.go

```diff
@@ -30,6 +31,9 @@ type channelManager struct {
 	pubsub *pubsub.PubSub
 
 	retransmissionTicker *retransmission.Ticker
+
+	relaySubscriptionsMutex sync.Mutex
+	relaySubscriptions      map[string]*pubsub.Subscription
```

Comment:
Maybe `forwarderSubscriptionsMutex` and `fowarderSubscriptions` to do not mix terms with threshold relay?